### PR TITLE
Add validation diagram and analyzers

### PR DESCRIPTION
**Pull request by twsouthwick**
This change adds a validation diagram to help maintain the architectural decisions so far and codify it into the build process.

### CODE REVIEW

**Comment by twsouthwick** [comment]
This implements the following diagram:

image

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Microsoft.DotNet.UpgradeAssistant.Steps.Packages.csproj [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
+    <!-- TODO: Enable once analysis is done -->
+    <NoWarn>$(NoWarn);DV0001</NoWarn>
```
Comment (mjrousos) [inline src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Microsoft.DotNet.UpgradeAssistant.Steps.Packages.csproj:5]: What is this for?

File: .vsconfig [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{
```
Comment (mjrousos) [inline .vsconfig:1]: What is this file used for?

**Review by mjrousos — APPROVED** [review]
Good change, thanks! There are a couple edited files whose purpose wasn't clear to me. Looks good otherwise.

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Microsoft.DotNet.UpgradeAssistant.Steps.Packages.csproj (hunk 1 above)
Comment (twsouthwick) [reply src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Microsoft.DotNet.UpgradeAssistant.Steps.Packages.csproj:5]: @sunandabalu introduced a dependency here that is broken; she's fixed it in a separate PR that is currently open. This disables that check

File: .vsconfig (hunk 2 above)
Comment (twsouthwick) [reply .vsconfig:1]: VS will now check for the workloads (ie validation tools) on startup and offer to install it if you don't have it.

**Pull request merged**